Does not compile with GCC

Hi there,

Any plans to make this plugin compatible with gcc?

```
$ LANG=C gcc -std=c++11 -shared -fPIC *.cpp -I../include -o libbm3d.so
In file included from ../include/Helper.h:30:0,
                 from ../include/BM3D_Base.h:25,
                 from BM3D_Base.cpp:20:
../include/Type.h:26:22: fatal error: xtr1common: No such file or directory
compilation terminated.
In file included from ../include/Helper.h:30:0,
                 from ../include/BM3D_Base.h:25,
                 from ../include/BM3D_Basic.h:24,
                 from BM3D_Basic.cpp:20:
../include/Type.h:26:22: fatal error: xtr1common: No such file or directory
compilation terminated.
In file included from ../include/Helper.h:30:0,
                 from ../include/BM3D_Base.h:25,
                 from ../include/BM3D_Final.h:24,
                 from BM3D_Final.cpp:20:
../include/Type.h:26:22: fatal error: xtr1common: No such file or directory
compilation terminated.
In file included from ../include/Helper.h:30:0,
                 from ../include/BM3D_Base.h:25,
                 from ../include/BM3D_Basic.h:24,
                 from ../include/BM3D.h:24,
                 from VSPlugin.cpp:20:
../include/Type.h:26:22: fatal error: xtr1common: No such file or directory
compilation terminated.

```

--- a/include/BM3D_Base.h
+++ b/include/BM3D_Base.h
@@ -148,14 +148,8 @@ private:
     template < typename _Ty >
     void process_core_gray();
 
-    template <>
-    void process_core_gray<FLType>();
-
     template < typename _Ty >
     void process_core_yuv();
-
-    template <>
-    void process_core_yuv<FLType>();
 
     template < typename _Ty >
     void process_core_rgb();

--- a/include/BM3D_Base.hpp
+++ b/include/BM3D_Base.hpp
@@ -82,7 +82,7 @@ void BM3D_Process_Base::process_core_gray()
 }
 
 template <>
-void BM3D_Process_Base::process_core_gray<FLType>()
+inline void BM3D_Process_Base::process_core_gray<FLType>()
 {
     // Get write/read pointer
     auto dstY = reinterpret_cast<FLType *>(vsapi->getWritePtr(dst, 0));
@@ -174,7 +174,7 @@ void BM3D_Process_Base::process_core_yuv()
 }
 
 template <>
-void BM3D_Process_Base::process_core_yuv<FLType>()
+inline void BM3D_Process_Base::process_core_yuv<FLType>()
 {
     // Get write/read pointer
     auto dstY = reinterpret_cast<FLType *>(vsapi->getWritePtr(dst, 0));

--- a/include/Block.h
+++ b/include/Block.h
@@ -1288,7 +1288,7 @@ void Block_Transform(_Dt1 &dst, const _St1 &src, _Fn1 &&_Func)
 
     for (auto upper = dstp + dst.size(); dstp != upper; ++dstp, ++srcp)
     {
-        *datap = _Func(*datap);
+        *dstp = _Func(*srcp);
     }
 }
 

--- a/include/Type.h
+++ b/include/Type.h
@@ -435,32 +435,32 @@ struct KeyPair
 
     bool operator==(const _Myt &_Right)
     {
-        return first == _Right.first;
+        return this->first == _Right.first;
     }
 
     bool operator!=(const _Myt &_Right)
     {
-        return first != _Right.first;
+        return this->first != _Right.first;
     }
 
     bool operator<(const _Myt &_Right)
     {
-        return first < _Right.first;
+        return this->first < _Right.first;
     }
 
     bool operator>(const _Myt &_Right)
     {
-        return first > _Right.first;
+        return this->first > _Right.first;
     }
 
     bool operator<=(const _Myt &_Right)
     {
-        return first <= _Right.first;
+        return this->first <= _Right.first;
     }
 
     bool operator>=(const _Myt &_Right)
     {
-        return first >= _Right.first;
+        return this->first >= _Right.first;
     }
 };
 

--- a/include/VAggregate.h
+++ b/include/VAggregate.h
@@ -81,14 +81,8 @@ private:
     template < typename _Dt1 >
     void process_core_gray();
 
-    template <>
-    void process_core_gray<FLType>();
-
     template < typename _Dt1 >
     void process_core_yuv();
-
-    template <>
-    void process_core_yuv<FLType>();
 
 protected:
     virtual void process_coreS() override

--- a/include/VAggregate.hpp
+++ b/include/VAggregate.hpp
@@ -79,7 +79,7 @@ void VAggregate_Process::process_core_gray()
 }
 
 template <>
-void VAggregate_Process::process_core_gray<FLType>()
+inline void VAggregate_Process::process_core_gray<FLType>()
 {
     std::vector<const FLType *> ResNumY, ResDenY;
 
@@ -152,7 +152,7 @@ void VAggregate_Process::process_core_yuv()
 }
 
 template <>
-void VAggregate_Process::process_core_yuv<FLType>()
+inline void VAggregate_Process::process_core_yuv<FLType>()
 {
     std::vector<const FLType *> ResNumY, ResDenY;
     std::vector<const FLType *> ResNumU, ResDenU;

--- a/include/VBM3D_Base.h
+++ b/include/VBM3D_Base.h
@@ -27,7 +27,7 @@
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 
-const struct VBM3D_Para
+struct VBM3D_Para
     : public BM3D_Para
 {
     typedef VBM3D_Para _Myt;
@@ -180,14 +180,8 @@ private:
     template < typename _Ty >
     void process_core_gray();
 
-    template <>
-    void process_core_gray<FLType>();
-
     template < typename _Ty >
     void process_core_yuv();
-
-    template <>
-    void process_core_yuv<FLType>();
 
     template < typename _Ty >
     void process_core_rgb();

--- a/include/VBM3D_Base.hpp
+++ b/include/VBM3D_Base.hpp
@@ -96,7 +96,7 @@ void VBM3D_Process_Base::process_core_gray()
 }
 
 template <>
-void VBM3D_Process_Base::process_core_gray<FLType>()
+inline void VBM3D_Process_Base::process_core_gray<FLType>()
 {
     std::vector<FLType *> dstYv;
     std::vector<const FLType *> srcYv;
@@ -229,7 +229,7 @@ void VBM3D_Process_Base::process_core_yuv()
 }
 
 template <>
-void VBM3D_Process_Base::process_core_yuv<FLType>()
+inline void VBM3D_Process_Base::process_core_yuv<FLType>()
 {
     std::vector<FLType *> dstYv;
     std::vector<FLType *> dstUv;

